src/main/java/edu/cornell/kfs/module/purap/document/CuPaymentRequestDocument.java

```diff
@@ -149,6 +147,26 @@ public void populateDocumentForRouting() {
     	}
     }
     
+    /* Cornell Customization : KFSPTS-34015
+     * We needed local fixes due to changes introduced by the KualiCo 2023-04-19 Upgrade patch changes.
+     * 
+     * In that patch, method org.kuali.kfs.module.purap.document.PaymentRequestDocument.doRouteStatusChange 
+     * introduced the following service call
+     *       getPdpExtractService().extractPaymentRequestDocument(this);
+     * which saves the paymentRequestDocument to record the extracted and paid timestamps resulting in 
+     * cash offset GL entries being duplicated.
+     * This GL duplicate entry issue was because method populateDocumentForRouting is invoked after a 
+     * docuemnt is saved. This check was added to the method so that populateDocumentForRouting prevented

```

Comment:
Minor typo in comment: `docuemnt` --> `document`

Reply:
fixed